src/components/shared/passphraseRenderer/index.js

```diff
@@ -0,0 +1,226 @@
+import React from 'react';
+import fillWordsList from 'bitcore-mnemonic/lib/words/english';
+import grid from 'flexboxgrid/dist/flexboxgrid.css';
+import { withTranslation } from 'react-i18next';
+import styles from './passphraseRenderer.css';
+import { PrimaryButton, TertiaryButton } from '../../toolbox/buttons/button';
+
+class PassphraseRenderer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      indexes: [2, 9],
+      fieldSelected: undefined,
+      chosenWords: {},
+      options: {},
+      isCorrect: false,
+      hasErrors: false,
+      disabledButton: true,
+      values: [],
+    };
+
+    this.handleConfirm = this.handleConfirm.bind(this);
+  }
+
+  componentDidMount() {
+    const { indexes } = this.state;
+    const values = this.props.passphrase.split(' ');
+
+    const options = this.assembleWordOptions(values, indexes);
+    this.setState({
+      ...this.state,
+      options: {
+        [indexes[0]]: options[0],
+        [indexes[1]]: options[1],
+      },
+      values,
+      fieldSelected: indexes[0],
+    });
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.timeout);
+  }
+
+  handleConfirm() {
+    const { chosenWords, indexes } = this.state;
+    const { values } = this.state;
+
+    const answers = Object.values(chosenWords);
+    const isCorrect = answers.filter((answer, index) => answer === values[indexes[index]])
+      .length === 2;
+
+    const cb = isCorrect
+      ? () => this.props.nextStep()
+      : () => this.setRandomIndexesFromPassphrase(2);
+
+    this.setState({
+      isCorrect,
+      hasErrors: !isCorrect,
+    });
+
+    this.timeout = setTimeout(cb, 1500);
+  }
+
+  setRandomIndexesFromPassphrase(qty) {
+    const { values } = this.state;
+    let idxs = values.map((w, index) => index);
+    const indexes = [...Array(qty)]
```

Comment:
the `qty` variable name is not very descriptive, could be something like:
```suggestion
    const indexes = [...Array(numberOfMissingWords)]
```

Reply:
✅